### ISSUE TITLE
`last()` not working as expected

### ISSUE THREAD

**Issue by awfulstew** [goto100/xpath#135]
When using `last()` as a filter, I would expect that the returned node would be the last one as ordered by location in the XML document (i.e. the ordering coming out of [`applyLocationPath`](https://github.com/goto100/xpath/blob/0617ceffaa782f76c39df8478b5ab505edeb4e2f/xpath.js#L2116)). The [way that predicates are applied](https://github.com/goto100/xpath/blob/0617ceffaa782f76c39df8478b5ab505edeb4e2f/xpath.js#L2112), however, is on an unsorted array which can (and in my case did) lead to the a different node (i.e. an earlier one) being returned.

**Comment by awfulstew** [comment]
This issue was happening on `0.0.32` of the library. Let me know if further details here would be helpful.

**Comment by awfulstew** [comment]
This is a shortened version of the file I was working with that still reproduces the issue that I am talking about:
```
<html>
  <body>
    <div>
      <blockquote type="cite" class="front-blockquote">
        <div id="1234">
          <span style="font-family: 'Times New Roman', serif; font-size: 16px; line-height: normal;">
            This is a test!
          </span>
        </div>
      </blockquote>
    </div>
    <div><br></div>
    <blockquote style="margin-Top: 0px; margin-Bottom: 0px; margin-Left: 0.5em">
      <blockquote type="cite" class="front-blockquote" style="margin-top: 0px;">
        <blockquote type="cite" class="front-blockquote" style="margin-top: 0px;">
          <blockquote type="cite" class="front-blockquote" style="margin-top: 0px;">
            <blockquote type="cite" class="front-blockquote" style="margin-top: 0px;">
              <blockquote type="cite" class="front-blockquote" style="margin-top: 0px;">
                <blockquote type="cite" class="front-blockquote" style="margin-top: 0px;">
                  <blockquote type="cite" class="front-blockquote" style="margin-top: 0px;">
                    <blockquote type="cite" class="front-blockquote" style="margin-top: 0px;">
                      <blockquote type="cite" class="front-blockquote" style="margin-top: 0px;">
                        <blockquote>
                          <span style="font-family: 'Times New Roman', serif; font-size: 16px; line-height: normal;">
                            This is also a test!
                          </span>
                        </blockquote>
                      </blockquote>
                    </blockquote>
                  </blockquote>
                </blockquote>
              </blockquote>
            </blockquote>
          </blockquote>
        </blockquote>
      </blockquote>
    </blockquote>
  </body>
</html>
```

Running the path expression `(.//*[local-name(.)='blockquote'])[not(@class="gmail_quote") and not(ancestor::blockquote)][last()]` will return the `This is a test` block rather than the expected top level of the `This is also a test` block.

**Comment by JLRishe** [comment]
Thank you for reporting this. This wasn't actually an issue with the last() function but rather an issue with the handling of context positions. Should now be fixed in version 0.0.34.
